examples/nlp/run_glue_with_BERT.py

```diff
@@ -0,0 +1,251 @@
+                                    data_dir=args.data_dir,
+                                    max_seq_length=args.max_seq_length,
+                                    processor=task_processors[0],
+                                    output_mode=output_mode,
+                                    evaluate=False,
+                                    **data_layer_params)
+
+nf.logger.info("Loading eval data...")
+eval_datasets = []
+
+# 2 task_processors for MNLI task: matched and mismatched dev set
+for task_processor in task_processors:
+    eval_datasets.append(nemo_nlp.GLUEDataset(
+                                            tokenizer=tokenizer,
+                                            data_dir=args.data_dir,
+                                            max_seq_length=args.max_seq_length,
+                                            processor=task_processor,
+                                            output_mode=output_mode,
+                                            evaluate=True,
+                                            **data_layer_params))
+
+# uses [CLS] token for classification (the first token)
+classifier = nemo_nlp.SequenceClassifier(hidden_size=hidden_size,
+                                         num_classes=num_labels,
+                                         log_softmax=False)
+if args.task_name == 'sts-b':
+    glue_loss = MSELoss()
+else:
+    glue_loss = CrossEntropyLoss()
+
+
+def create_pipeline(dataset, batch_size=args.batch_size,
+                    local_rank=args.local_rank, num_gpus=args.num_gpus):
+
+    data_layer = nemo_nlp.BertSentenceClassificationDataLayer(
+            dataset=dataset,
+            batch_size=batch_size,
+            num_workers=0,
+            factory=nf)
+
+    input_ids, input_type_ids, input_mask, labels = data_layer()
+
+    hidden_states = model(input_ids=input_ids,
+                          token_type_ids=input_type_ids,
+                          attention_mask=input_mask)
+    logits = classifier(hidden_states=hidden_states)
+
+    loss = glue_loss(logits=logits, labels=labels)
+
+    steps_per_epoch = len(data_layer) // (batch_size * num_gpus)
+    return loss, steps_per_epoch, data_layer, [logits, labels]
+
+
+train_loss, steps_per_epoch, _, _ = create_pipeline(train_dataset)
+_, _, eval_data_layer, eval_tensors = \
+                               create_pipeline(eval_datasets[0])
+
+callbacks_eval = [nemo.core.EvaluatorCallback(
+    eval_tensors=eval_tensors,
+    user_iter_callback=lambda x, y: eval_iter_callback(
+        x, y, eval_data_layer, output_mode),
+    user_epochs_done_callback=lambda x: eval_epochs_done_callback(
+        x, args.work_dir, eval_task_names[0]),
+    tb_writer=nf.tb_writer,
+    eval_step=steps_per_epoch)]
+
+# create additional callback and data layer for MNLI mismatched dev set
+if args.task_name == 'mnli':
+    _, _, eval_data_layer_mm, eval_tensors_mm = \
+                               create_pipeline(eval_datasets[1])
```

Comment:
we should find a way to be able to evaluate all eval datasets, not just hardcode 0 and 1